pythclient/price_feeds.py

```diff
@@ -0,0 +1,430 @@
+    )
+    offset += 8
+
+    last_attested_publish_time = int.from_bytes(
+        bytes_[offset : offset + 8], byteorder="big", signed=True
+    )
+    offset += 8
+
+    return {
+        "product_id": product_id,
+        "price_id": price_id,
+        "price": str(price),
+        "conf": str(conf),
+        "expo": expo,
+        "ema_price": str(ema_price),
+        "ema_conf": str(ema_conf),
+        "status": status,
+        "num_publishers": num_publishers,
+        "max_num_publishers": max_num_publishers,
+        "attestation_time": attestation_time,
+        "publish_time": publish_time,
+        "prev_publish_time": prev_publish_time,
+        "prev_price": str(prev_price),
+        "prev_conf": str(prev_conf),
+        "last_attested_publish_time": last_attested_publish_time,
+    }
+
+
+def vaa_to_price_info(price_feed_id, vaa) -> PriceInfo:
+    parsed_vaa = parse_vaa(vaa)
+
+    try:
+        batch_attestation = parse_batch_price_attestation(parsed_vaa["payload"])
+    except Exception as e:
+        logging.error(e)
+        logging.error(f"Parsing historical VAA failed: {parsed_vaa}")
+        return None
+
+    for price_attestation in batch_attestation["price_attestations"]:
+        if price_attestation["price_id"] == price_feed_id:
+            return create_price_info(
+                price_attestation,
+                vaa,
+                parsed_vaa["sequence"],
+                parsed_vaa["emitter_chain"],
+            )
+
+    return None
+
+
+def create_price_info(price_attestation, vaa, sequence, emitter_chain):
+    price_feed = price_attestation_to_price_feed(price_attestation)
+    return PriceInfo(
+        seq_num=int(sequence),
+        vaa=vaa,
+        publish_time=price_attestation["publish_time"],
+        attestation_time=price_attestation["attestation_time"],
+        last_attested_publish_time=price_attestation["last_attested_publish_time"],
+        price_feed=price_feed,
+        emitter_chain_id=emitter_chain,
+    )
+
+
+def price_attestation_to_price_feed(price_attestation):
+    ema_price = Price(
+        price_attestation["ema_conf"],
+        price_attestation["expo"],
+        price_attestation["ema_price"],
+        price_attestation["publish_time"],
+    )
+
+    if price_attestation["status"] == 1:  # Assuming 1 means Trading
+        price = Price(
+            price_attestation["conf"],
+            price_attestation["expo"],
+            price_attestation["price"],
+            price_attestation["publish_time"],
+        )
+    else:
+        price = Price(
+            price_attestation["prev_conf"],
+            price_attestation["expo"],
+            price_attestation["prev_price"],
+            price_attestation["prev_publish_time"],
+        )
+        ema_price.publish_time = price_attestation["prev_publish_time"]
+
+    return PriceFeed(ema_price, price_attestation["price_id"], price)
+
+
+id = "e62df6c8b4a85fe1a67db44dc12de5db330f7ac66b72dc658afedf0f4a415b43"
```

Comment:
seems like this logic was supposed to be deleted?

Reply:
ah yes thanks